aws/transit_gateway_test.go

```diff
@@ -0,0 +1,294 @@
+
+	//TransitGateway takes some time to be available and there isn't Waiters available yet
+	//To avoid test errors, I'm introducing a sleep call
+	time.Sleep(180 * time.Second)
+	tgwRouteTable := *result.TransitGatewayRouteTable
+	return tgwRouteTable
+}
+
+func TestGetAllTransitGatewayRouteTableInstances(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	session, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region)},
+	)
+	require.NoError(t, err)
+
+	tgwRouteTableName := "clud-nuke-test-" + util.UniqueID()
+	tgwRouteTable := createTestTransitGatewayRouteTable(t, session, tgwRouteTableName)
+
+	defer nukeAllTransitGatewayRouteTables(session, []*string{tgwRouteTable.TransitGatewayRouteTableId})
+	defer nukeAllTransitGatewayInstances(session, []*string{tgwRouteTable.TransitGatewayId})
+
+	ids, err := getAllTransitGatewayRouteTables(session, region, time.Now().Add(1*time.Hour*-1))
+	require.NoError(t, err)
+	assert.NotContains(t, awsgo.StringValueSlice(ids), awsgo.StringValue(tgwRouteTable.TransitGatewayRouteTableId))
+
+	ids, err = getAllTransitGatewayRouteTables(session, region, time.Now().Add(1*time.Hour))
+	require.NoError(t, err)
+	assert.Contains(t, awsgo.StringValueSlice(ids), awsgo.StringValue(tgwRouteTable.TransitGatewayRouteTableId))
+}
+
+func TestNukeTransitGatewayRouteTable(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+
+	session, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region)},
+	)
+	require.NoError(t, err)
+
+	svc := ec2.New(session)
+
+	tgwRouteTableName := "clud-nuke-test-" + util.UniqueID()
+	tgwRouteTable := createTestTransitGatewayRouteTable(t, session, tgwRouteTableName)
+	defer nukeAllTransitGatewayInstances(session, []*string{tgwRouteTable.TransitGatewayId})
+
+	_, err = svc.DescribeTransitGatewayRouteTables(&ec2.DescribeTransitGatewayRouteTablesInput{
+		TransitGatewayRouteTableIds: []*string{
+			tgwRouteTable.TransitGatewayRouteTableId,
+		},
+	})
+	require.NoError(t, err)
+
+	err = nukeAllTransitGatewayRouteTables(session, []*string{tgwRouteTable.TransitGatewayRouteTableId})
+	require.NoError(t, err)
+
+	ids, err := getAllTransitGatewayRouteTables(session, region, time.Now().Add(1*time.Hour))
+	require.NoError(t, err)
+	assert.NotContains(t, awsgo.StringValueSlice(ids), awsgo.StringValue(tgwRouteTable.TransitGatewayRouteTableId))
+}
+
+func createTestTransitGatewayVpcAttachment(t *testing.T, session *session.Session, name string) ec2.TransitGatewayVpcAttachment {
+	svc := ec2.New(session)
+
+	transitGateway := createTestTransitGateway(t, session, name)
+
+	input := &ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   awsgo.String("isDefault"),
+				Values: []*string{awsgo.String("true")},
+			},
+		},
+	}
+
+	vpcs, _ := svc.DescribeVpcs(input)
```

Comment:
We probably do want an error check here because the default VPC may actually not be present (cloud-nuke has a feature to remove default VPCs).

Reply:
Great suggestion! Implemented!

Could you check again, please?